pkg/api/validate.go

```diff
@@ -174,7 +245,7 @@ func validateAgentPoolProfiles(apps []AgentPoolProfile) (errs []error) {
 	return
 }
 
-func validateAgentPoolProfile(app *AgentPoolProfile) (errs []error) {
+func validateAgentPoolProfile(app AgentPoolProfile, vnet *net.IPNet) (errs []error) {
```

Comment:
why is `AgentPoolProfile` changing?  Seems appropriate to pass as a pointer rather than a copy.

Reply:
It seemed to make sense on Friday but I think I agree with you on Monday.  I'll fix it up.

Reply:
Ah, the Friday reasoning was, that doing it your way would introduce a nil check for something that cannot possibly be nil in the actual structure as passed in by the user - so you end up with a synthetic theoretically-user-facing error message for something that can only be caused by a logic fault by the programmer.  That's why I removed the pointer.  Thoughts on what to do @pweil- ?

Reply:
if it is validated as non-nil elsewhere then we shouldn't have to worry about the check.  There are other examples of it in this PR (ie. oldCS can never be nil when it gets to the internal update validation method, it's checked in the public one) but they're just nits IMO.  It doesn't hurt either way, to me.  

I lean on the side of being more defensive, even if it shouldn't happen, so long as it's not making the code hard to read.